taichi/ir/type_utils.cpp

```diff
@@ -53,6 +53,36 @@ int data_type_size(DataType t) {
   }
 }
 
+std::string tensor_type_format_helper(const std::vector<int> &shape,
+                                      std::string format_str,
+                                      int dim) {
+  std::string fmt = "[";
+  for (int i = 0; i < shape[dim]; ++i) {
+    if (dim != shape.size() - 1) {
+      fmt += tensor_type_format_helper(shape, format_str, dim + 1);
+    } else {
+      fmt += format_str;
+    }
+    if (i != shape[dim] - 1) {
+      fmt += ", ";
+      if (dim == 0) {
+        fmt += "\n";
```

Comment:
Do we need this when the tensor is a vector instead of a matrix?

Reply:
Good point. Let's keep 1-line for Vectors.